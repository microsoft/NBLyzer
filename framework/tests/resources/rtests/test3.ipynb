{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#读取数据集\n",
    "from tensorflow.examples.tutorials.mnist import input_data\n",
    "mnist_data_folder='/home/feier/python3/tensorflow/'\n",
    "mnist=input_data.read_data_sets(mnist_data_folder,one_hot=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#softmax\n",
    "x = tf.placeholder(\"float\",[None, 784])\n",
    "\n",
    "W = tf.Variable(tf.zeros([784,10]))\n",
    "b = tf.Variable(tf.zeros([10]))\n",
    "\n",
    "y = tf.nn.softmax(tf.matmul(x, W)+b)\n",
    "\n",
    "y_ = tf.placeholder(tf.float32, [None, 10])\n",
    "\n",
    "cross_entropy = tf.reduce_mean(-tf.reduce_sum(y_ *tf.log(y), reduction_indices=[1]))\n",
    "\n",
    "train_step = tf.train.GradientDescentOptimizer(0.01).minimize(cross_entropy)\n",
    "\n",
    "init = tf.initialize_all_variables()\n",
    "\n",
    "sess = tf.Session()\n",
    "sess.run(init)\n",
    "for i in range(10000):\n",
    "    batch_xs, batch_ys = input_data.train.next_batch(100)\n",
    "    sess.run(train_step, feed_dict={x:batch_xs, y_:batch_ys})\n",
    "\n",
    "correct_prediction = tf.equal(tf.argmax(y,1), tf.argmax(y_,1))\n",
    "accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
