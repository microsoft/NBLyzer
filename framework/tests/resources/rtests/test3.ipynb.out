[{"cell_id":0,"errors":[{"line":3,"label":"mnist_data_folder", "error_type":"ErrorType.TERMINAL", "message":"Isolated cell."},{"line":2,"label":"mnist", "error_type":"ErrorType.TERMINAL", "message":"Isolated cell."}],"path":[0]},{"cell_id":0,"errors":[{"line":3,"label":"mnist_data_folder", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":2,"label":"mnist", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[0]},{"cell_id":1,"errors":[{"line":1,"label":"x", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":4,"label":"W", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":4,"label":"b", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":7,"label":"y", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":9,"label":"y_", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":11,"label":"cross_entropy", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":13,"label":"train_step", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":15,"label":"init", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":17,"label":"sess", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":20,"label":"batch_xs", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":20,"label":"batch_ys", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":23,"label":"correct_prediction", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."},{"line":24,"label":"accuracy", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[1]}]